monitoring/prometheus/alerts/ceph_default_alerts.yml

```
@@ -230,8 +230,7 @@ groups:
       - alert: pool filling up
         expr: |
           (
-            predict_linear(ceph_pool_stored[2d], 3600 * 24 * 5) >=
-            ceph_pool_max_avail
+            predict_linear(ceph_pool_max_avail[2d], 3600 * 24 * 5)  < 0
```

Thanks for your contribution! It seems you're right about the problem of the alert.

Though, I wonder if an alert purely based on the `ceph_pool_max_avail` metric would always be triggered by the pool that was filled up and therefore led to the reduction of the available space. I think, it might rather be that the pool with the highest usage would be reported first, when the alert is based on the `ceph_pool_max_avail` metric only.

We may be able to work around that by keeping the `ceph_pool_stored` metric as the one to be predicted. By summing up the `ceph_pool_stored` and `ceph_pool_max_avail` metrics, we should be able to compare them properly with the predicted value.

What do you think?

```suggestion
            predict_linear(ceph_pool_stored[2d], 3600 * 24 * 5)
            >= ceph_pool_stored + ceph_pool_max_avail
```

---

> Though, I wonder if an alert purely based on the ceph_pool_max_avail metric would always be triggered by the pool that was filled up and therefore led to the reduction of the available space. I think, it might rather be that the pool with the highest usage would be reported first, when the alert is based on the ceph_pool_max_avail metric only.

It's tricky indeed. The `ceph_pool_max_avail` is the same across all pools with the same crush ruleset, but `ceph_pool_stored` is pool specific. If you think your suggestion prevents another problem then I'll gladly update my commit 👍